### PR TITLE
Fix NOTICE file content

### PR DESCRIPTION
**Pull request by jbonofre**
Updating NOTICE file with:
- SGA
- include NOTICE file from ALv2 used code

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -4,11 +4,19 @@ Copyright 2024 The Apache Software Foundation
...
-* Creative Commons 1.0 Universal
-* Eclipse Distribution License, Version 1.0
-* Eclipse Public License, Version 1.0
-* Eclipse Public License, Version 2.0
-* GNU General Public License, Version 2 with the GNU Classpath Exception
+The initial code for the Polaris project was donated
+to the ASF by Snowflake Inc. (https://www.snowflake.com/) copyright 2024.
+
+Apache Iceberg
+Copyright 2017-2022 The Apache Software Foundation
```
Comment (snazy) [inline NOTICE:11]: Do we really need the year(s) in the file?
Why not `and onwards` here as well?

Comment (jbonofre) [reply NOTICE:11]: We don't have the choice: we have to include the NOTICE file content as it is from ALv2 code used. So it's directly the content of Iceberg NOTICE file.

**Review by snazy — APPROVED** [review]

File: NOTICE [hunk 2]
```diff
@@ -4,11 +4,19 @@ Copyright 2024 The Apache Software Foundation
...
+Apache Spark
+Copyright 2014 and onwards The Apache Software Foundation.
+
+Apache Hadoop
+Copyright 2006 and onwards The Apache Software Foundation.
+
+Dropwizard
+Copyright 2010-2013 Coda Hale and Yammer, Inc., 2014-2020 Dropwizard Team
+
+This product includes software developed by Coda Hale and Yammer, Inc.
```
Comment (adutra) [inline NOTICE:22]: Shouldn't this sentence be at the top, close to "This product includes software developed at The Apache Software Foundation"?

Comment (jbonofre) [reply NOTICE:22]: Order doesn't matter. It's the drop wizard notice content.

**Comment by jbonofre** [comment]
NB: it's according to asf legal and ALv2 4.d section

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
```
* What went wrong:
Execution failed for task ':generatePomFileForMavenPublication'.
> Unable to process url: https://projects.apache.org/json/projects/polaris.json
```

**Comment by jbonofre** [comment]
@RussellSpitzer yup, I saw that, I will investigate.

**Pull request merged**